### PR TITLE
Invalidate Shapes BackgroundColor

### LINKED ISSUE #18131: [iOS, Mac, Windows] Color changes are not reflected in the Rectangle shapes
### Description

When changing the color of Rectangle shape at the run time, the change is not reflected in iOS, Mac and Windows.

### Steps to Reproduce

1. Run the attached sample in iOS, Mac or Windows.
2. Click the Add Rectangle button. It have the code to add a Rectangle in Absolute Layout with a primary color. 
3. Click Change Color. It have code changes to change the color of the Rectangle.
You can see the color change is not applied properly.

[RectangleColorChange.zip](https://github.com/dotnet/maui/files/13041892/RectangleColorChange.zip)

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-rc.2.9373

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

6.0

### Affected platforms

iOS, Windows, macOS

### Affected platform versions

iOS 16.4 (simulator)

### Did you find any workaround?

No

### Relevant log output

```shell
NA
```

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix invalidating Shapes BackgroundColor.

|Android | Windows |
|:-:|:-:|
|fix-18131-droid|fix-18131-win|

### Issues Fixed

Fixes #18131

### CODE REVIEW

**Review by mattleibow — COMMENTED** [review]
Will it be valuable to add another view - ellipse - to the test?

**Comment by jsuarezruiz** [comment]
> Will it be valuable to add another view - ellipse - to the test?

Sure thing added the Ellipse too.

**Comment by jsuarezruiz** [comment]
We can set the background of Shapes with BackgroundColor, Background and Fill. In the case of the Rectangle, everything is applied to the background of the Shape equally. **Fill > Background > BackgroundColor**, Fill overwrites Background and if Background is used it overwrites BackgroundColor.
With an Ellipse, setting Background or BackgroundColor applies to the Shape's background; but if BackgroundColor or Background is applied in combination with Fill, Fill will fill the Shape and BackgroundColor is applied to the background color of the View.

image

File: src/Core/src/Handlers/ShapeView/ShapeViewHandler.Windows.cs [hunk 1]
```diff
@@ -14,15 +13,13 @@ protected override W2DGraphicsView CreatePlatformView()
 
 		public static void MapBackground(IShapeViewHandler handler, IShapeView shapeView)
 		{
+			handler.UpdateValue(nameof(IViewHandler.ContainerView));
+			handler.ToPlatform().UpdateBackground(shapeView);
+
 			// If Fill and Background are not null, will use Fill for the Shape background
 			// and Background for the ShapeView background.
 			if (shapeView.Background is not null && shapeView.Fill is not null)
```
Comment (emaf) [inline src/Core/src/Handlers/ShapeView/ShapeViewHandler.Windows.cs:21]: Is this if statement and the comment above still valid?

**Review by emaf — DISMISSED** [review]

**Pull request closed without merging**